micrometer-observation-test/src/main/java/io/micrometer/observation/tck/ObservationValidator.java

```diff
@@ -35,14 +33,12 @@
  */
 class ObservationValidator implements ObservationHandler<Context> {
 
-    private static final InternalLogger LOGGER = InternalLoggerFactory.getInstance(ObservationValidator.class);
-
     private final Consumer<ValidationResult> consumer;
 
     private final Predicate<Context> supportsContextPredicate;
 
     ObservationValidator() {
-        this(validationResult -> LOGGER.warn(validationResult.toString()));
+        this(ObservationValidator::throwInvalidObservationException);
```

Comment:
How hard would it be to update the tests to assert on the exception (or lack thereof) rather than using a custom consumer that users can't themselves use?

Reply:
That's a great idea and I have no clue why I did not do that way originally. :)